wagtail/wagtailcore/wagtail_hooks.py

```diff
@@ -1,8 +1,17 @@
 from __future__ import absolute_import, unicode_literals
 
+from django.conf import settings
+from django.contrib.auth.views import redirect_to_login
 from django.core.urlresolvers import reverse
 
+from wagtail.utils.compat import user_is_authenticated
 from wagtail.wagtailcore import hooks
+from wagtail.wagtailcore.models import PageViewRestriction
+
+
+def require_wagtail_login(next):
```

Comment:
Might users want to override `settings.WAGTAIL_FRONTEND_LOGIN_URL` on a per-section basis?